xarray/backends/netCDF4_.py

```diff
@@ -210,9 +216,9 @@ def __init__(self, filename, mode='r', format='NETCDF4', group=None,
         self.ds = opener()
```

Comment:
Actually, `netCDF4.Dataset` has a `__reduce__` method that raises NotImplementedError, so maybe this isn't necessary